src/internal_modules/roc_fec/block_writer.cpp

```diff
@@ -46,9 +46,14 @@ BlockWriter::BlockWriter(const BlockWriterConfig& config,
         return;
     }
 
-    cur_sbn_ = (packet::blknum_t)core::fast_random_range(0, packet::blknum_t(-1));
-    cur_block_repair_sn_ =
-        (packet::seqnum_t)core::fast_random_range(0, packet::seqnum_t(-1));
+    bool ok = core::secure_random(&cur_sbn_, sizeof(cur_sbn_));
+    if (!ok) {
+        return;
+    }
+    ok = core::secure_random(&cur_block_repair_sn_, sizeof(cur_block_repair_sn_));
+    if (!ok) {
+        return;
+    }
```

Comment:
In case of error we should set init status (we have a code for prng failure)

Same in other ctors.